process/process.go

```diff
@@ -17,41 +17,109 @@ import (
 
 type processesFn func() ([]ps.Process, error)
 
-func findPIDsWithFn(fn processesFn, prefix string, log logging.Logger) ([]int, error) {
-	processes, err := fn()
+type matchFn func(ps.Process) bool
+
+func matchPath(p ps.Process, match string, log logging.Logger) bool {
```

Comment:
Refactored the match functions into a Matcher with match types (PathEqual/PathContains/...)